src/operator/nn/activation.cc

```diff
@@ -128,17 +130,25 @@ inline static bool BackwardActStorageType(const nnvm::NodeAttrs& attrs,
   bool ret = false;
   const ActivationParam& param = nnvm::get<ActivationParam>(attrs.parsed);
 #if (MXNET_USE_CUDNN == 1 || MXNET_USE_MKLDNN == 1)
-  if (param.act_type != activation::kReLU) {
-    CHECK_EQ(in_attrs->size(), 3U);
-    ret = ElemwiseStorageType<3, 1, false, false, false>(attrs, dev_mask,
-                                                         dispatch_mode,
-                                                         in_attrs, out_attrs);
+  bool should_continue = true;
+#if MXNET_USE_MKLDNN == 1
+  if (!(dev_mask == mshadow::cpu::kDevMask && SupportMKLDNNAct(param))) {
+    should_continue = false;
+  }
+#endif
+  if (should_continue) {
+    if (param.act_type != activation::kReLU) {
+      CHECK_EQ(in_attrs->size(), 3U);
+      ret = ElemwiseStorageType<3, 1, false, false, false>(
+          attrs, dev_mask, dispatch_mode, in_attrs, out_attrs);
```

Comment:
I still have the same question here. dispatch_mode can be kFComputeEx? ElemwiseStorageType only uses kFComputeEx for sparse storage type.